src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,150 @@
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+class TodoList extends React.PureComponent {
+  state = {
+    search: '',
+    todoStatus: 'all',
+    random: null,
+  }
+
+  searchHandler = (event) => {
+    this.setState({
+      search: event.target.value,
+    });
+  }
+
+  searchFilter = (todos) => {
+    if (!todos) {
+      return [];
+    }
+
+    const { search } = this.state;
+    const searchToLower = search.toLowerCase();
+
+    return todos.filter(({ title }) => (
+      title && title.toLowerCase().includes(searchToLower)));
+  }
+
+  todoStatusHandler = (event) => {
+    this.setState({
+      todoStatus: event.target.value,
+    });
+  }
+
+  todosCompletedFilter = (todos) => {
+    const { todoStatus } = this.state;
+
+    switch (todoStatus) {
+      case 'all':
+        return todos;
+      case 'active':
+        return todos.filter(({ completed }) => !completed);
+      case 'inactive':
+        return todos.filter(({ completed }) => completed);
+      default:
+        return todos;
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  randomSort(todos) {
+    return [...todos].sort(() => 0.5 - Math.random());
+  }
+
+  render() {
+    const { todos, onChangeUser, selectedUserId } = this.props;
+    const { search, todoStatus, random } = this.state;
+    let todosForShow = this.searchFilter(todos);
+
+    todosForShow = this.todosCompletedFilter(todosForShow);
+    if (random) {
+      todosForShow = this.randomSort(todosForShow);
+    }
+
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
+
+        <input
+          type="text"
+          value={search}
+          onChange={this.searchHandler}
+          placeholder="search"
+        />
+
+        <select
+          name="todoStatus"
+          onChange={this.todoStatusHandler}
+          value={todoStatus}
+        >
+          <option value="all">All</option>
+          <option value="active">active</option>
+          <option value="inactive">completed</option>
+        </select>
+
+        <button
+          type="button"
+          onClick={() => this.setState({ random: Math.random() })}
+        >
+          randomSort
+        </button>
+
+        <div className="TodoList__list-container">
+          <ul className="TodoList__list">
+            {todosForShow.map(todo => (
+              <li
+                key={todo.id}
+                className={classNames(
+                  'TodoList__item',
+                  {
+                    'TodoList__item--unchecked': !todo.completed,
+                    'TodoList__item--checked': todo.completed,
+                  },
+                )}
```

Comment:
Я бы вот так отформатировал
```suggestion
                className={classNames('TodoList__item', {
                  'TodoList__item--unchecked': !todo.completed,
                  'TodoList__item--checked': todo.completed,
                })}
```